src/analyses/invariant_set_domain.cpp

```diff
@@ -28,12 +28,10 @@ void invariant_set_domaint::transform(
     {
       // Comparing iterators is safe as the target must be within the same list
       // of instructions because this is a GOTO.
-      exprt tmp(from_l->guard);
+      exprt tmp(from_l->get_condition());
 
-      goto_programt::const_targett next=from_l;
-      next++;
-      if(next==to_l)
-        tmp = boolean_negate(from_l->guard);
+      if(std::next(from_l) == to_l)
+        tmp = boolean_negate(from_l->get_condition());
```

Comment:
Since `get_condition()` may have non-trivial cost in future, this should be `tmp = boolean_negate(tmp);`